schema-templates/basic-auth.sql

```diff
@@ -0,0 +1,291 @@
+-------------------------------------------------------------------------------
+-- Adapted from https://github.com/robconery/pg-auth
+
+begin;
+
+create extension if not exists pgcrypto;
+create extension if not exists "uuid-ossp";
+create schema if not exists basic_auth;
+
+-------------------------------------------------------------------------------
+-- Utility functions
+
+create or replace function
+basic_auth.clearance_for_role(u name) returns void as
+$$
+declare
+  ok boolean;
+begin
+  select exists (
+    select rolname
+      from pg_authid
+     where pg_has_role(current_user, oid, 'member')
+       and rolname = u
+  ) into ok;
+  if not ok then
+    raise invalid_password using message =
+      'current user not member of role ' || u;
+  end if;
+end
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------------------------------
+-- Login storage and constraints
+
+create table if not exists
+basic_auth.logins (
+  username character varying not null,
+  pass   character(60) not null,
+  role   name not null,
+  email  character varying not null unique,
+  active boolean not null default false,
+  more   JSON,
```

Comment:
Since this is not directly used here, remove all together (replace with a comment like "you can add additional columns or a json column"). This is a sample structure and users will chose weather to add new separate columns or have a single json column for custom notes about users.